cardano/src/block/verify_chain.rs

```diff
@@ -0,0 +1,104 @@
+use block::*;
+use address;
+use config::{ProtocolMagic};
+
+pub struct ChainState {
+    pub prev_block: HeaderHash,
+    pub prev_date: Option<BlockDate>,
+    pub slot_leaders: Vec<address::StakeholderId>,
```

Comment:
I would also add the expected `protocol_magic` here as it is a parameter of the blockchain and it must be consistent within the whole blockchain.